### PR TITLE
Add `Lint/UnusedBlockArgument` rule

### LINKED ISSUE #158: New Rule: Block not captured
Anonymous block annotations are present since a bit of time now. Adding `&` shows that the method yields a block, but using `&block` either means that or that the block is captured.

Here is a simple example:
```cr
# Valid
def m(&block)
  block.call 1
end

# Also valid
def m(&block)
  callback = block
end

# Invalid
def m(&block)
  yield
end

# Correct syntax
def m(&)
  yield
end
```

### PR DESCRIPTION
**Pull request by Sija**
Resolves #158

### CODE REVIEW

File: spec/ameba/rule/lint/unused_block_argument_spec.cr [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      )
+      subject.catch(s).should be_valid
+    end
+
+    it "doesn't report if anonymous" do
+      s = Source.new %(
+        def method(a, b, c, &)
+        end
+      )
+      subject.catch(s).should be_valid
```
Comment (veelenga) [inline spec/ameba/rule/lint/unused_block_argument_spec.cr:22]: shouldn't we use that newer matcher `expect_no_issues`/`expect_issue` instead?

File: src/ameba/ast/scope.cr [hunk 2]
```diff
@@ -143,6 +146,14 @@ module Ameba::AST
       end || variable.used_in_macro?
     end
 
+    # Returns `true` if current scope (or any of inner scopes) references variable,
+    # `false` otherwise.
+    def yields?(check_inner_scopes = true)
```
Comment (veelenga) [inline src/ameba/ast/scope.cr:151]: this description is wrong

File: spec/ameba/rule/lint/unused_block_argument_spec.cr (hunk 1 above)
Comment (Sija) [reply spec/ameba/rule/lint/unused_block_argument_spec.cr:22]: yeah, we should but I was too lazy to rewrite those, xmas & all that jazz, sorry!

Comment (Sija) [reply spec/ameba/rule/lint/unused_block_argument_spec.cr:22]: There are still many specs to rewrite anyway...

**Review by veelenga — APPROVED** [review]
Lgtm

**Pull request merged**